Automattic/wp-calypso
# Track all user+site signup conversions in Google Analytics and AdWords

Ref: p9f5aF-1qH-p2

## Summary of Changes

- Removes support for tracking "free" signups in Google Analytics and AdWords.
- Adds support for tracking all user+site signups in Google Analytics and AdWords, because we're modifying our approach to one that tracks all new users that create a site. We'll then filter stats on the GA and AdWords end of things by cross-referencing those who also have a cart conversion.

## Testing Instructions

- Use an incognito window (make sure you're logged-out).

- Visit: [this Calypso live URL](https://hash-5c496f17b0500f3455cf08f5d8f046ed7e3ed4ea.calypso.live/start/about?flags=gdpr-banner,google-analytics,ad-tracking)

- Enable debug notes by entering the following in your browser console:
  ```
  localStorage.setItem( 'debug', 'calypso:analytics:*' );
  ```

- Also be sure to allow tracking by setting the cookie banner to `yes`:
  ```
  document.cookie = 'sensitive_pixel_option=yes;'
  ```

- Refresh and make sure you're (a) logged-out, (b) on this [this Calypso live URL](https://hash-5c496f17b0500f3455cf08f5d8f046ed7e3ed4ea.calypso.live/start/about?flags=gdpr-banner,google-analytics,ad-tracking), and (c) that when you filter the "Console" tab by `isAdTrackingAllowed` that you see all `true` values.

  <img width="596" alt="2018-08-21_08-55-04" src="https://user-images.githubusercontent.com/1563559/44416659-eb985b80-a51f-11e8-9c4a-db437e68161a.png">

- Leave the browser console open and create a new site and user account.

  <img width="1020" alt="2018-08-21_08-58-28" src="https://user-images.githubusercontent.com/1563559/44416804-6bbec100-a520-11e8-909e-97d3d4da7711.png">

- Choose the "Free" plan.
  <img width="765" alt="2018-08-21_08-56-10" src="https://user-images.githubusercontent.com/1563559/44416721-1b476380-a520-11e8-983a-008e054dc17d.png">

- Enter account details and click "Continue".
  <img width="454" alt="2018-08-21_09-00-12" src="https://user-images.githubusercontent.com/1563559/44416885-a7f22180-a520-11e8-89bc-fa1c337ea6f0.png">

---

- Now filter the "Console" tab by `record`. You should have entries like this. Note `is_signup: true`.
  <img width="933" alt="2018-08-24_07-05-09" src="https://user-images.githubusercontent.com/1563559/44592136-2b538300-a76c-11e8-93e4-0dab2101de39.png">

- Also filter the Network tab by `new-user-site` and confirm these four entries with a `200` or `302` response status.
  ![2018-09-26_08-26-47](https://user-images.githubusercontent.com/1563559/46094311-0df84700-c166-11e8-8782-682c26589bfd.jpg)



---

## Test a Paid Plan

- Please clear your console and repeat the first few steps. This time choose a paid plan. When you arrive at the checkout step, if `isAdTrackingAllowed` appears in console as `false`, add `?flags=gdpr-banner,google-analytics,ad-tracking` to the URL and refresh the page.

- Complete checkout and filter the console and network tab again. Confirm no instances of `new-user-site` (new slug) or `free-plan` (old slug) in the Console or Network tabs. Also filter by `ad-tracking` in Console tab and confirm that all existing cart conversion trackers continue to work for a paid plan.

In the line that reads "Recording purchase" the data reported should NOT include `is_signup: true`.

<img width="1128" alt="2018-08-24_07-15-31" src="https://user-images.githubusercontent.com/1563559/44592615-93ef2f80-a76d-11e8-955e-34922cc6ee1b.png">


## Other Flows (No Breakage)

- Test checkout with other products in your cart, such as a G Suite purchase, test Jetpack, test each of the WPCOM plans. Confirm no errors in browser console and that checkout continues to work.
